aws/resource_aws_macie2_classification_job_test.go

```diff
@@ -0,0 +1,514 @@
+					resource.TestCheckResourceAttr(resourceName, "s3_job_definition.0.bucket_definitions.0.buckets.0", bucketName),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_at"),
+				),
+			},
+			{
+				Config: testAccAwsMacieClassificationJobconfigStatus(bucketName, macie2.JobStatusUserPaused),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2ClassificationJobExists(resourceName, &macie2Output2),
+					testAccCheckAwsMacie2ClassificationJobNotRecreated(&macie2Output, &macie2Output2),
+					resource.TestCheckResourceAttr(resourceName, "job_type", macie2.JobTypeScheduled),
+					resource.TestCheckResourceAttr(resourceName, "job_status", macie2.JobStatusUserPaused),
+					resource.TestCheckResourceAttr(resourceName, "s3_job_definition.0.bucket_definitions.0.buckets.0", bucketName),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_at"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsMacie2ClassificationJob_complete(t *testing.T) {
+	var macie2Output macie2.DescribeClassificationJobOutput
+	resourceName := "aws_macie2_classification_job.test"
+	bucketName := "test-bucket-name-aws"
+	description := "Description of a test"
+	descriptionUpdated := "Updated Description of a test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckAwsMacie2ClassificationJobDestroy,
+		ErrorCheck:        testAccErrorCheck(t, macie2.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsMacieClassificationJobconfigComplete(bucketName, macie2.JobStatusRunning, description),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2ClassificationJobExists(resourceName, &macie2Output),
+					resource.TestCheckResourceAttr(resourceName, "job_type", macie2.JobTypeScheduled),
+					resource.TestCheckResourceAttr(resourceName, "job_status", macie2.JobStatusRunning),
+					resource.TestCheckResourceAttr(resourceName, "s3_job_definition.0.bucket_definitions.0.buckets.0", bucketName),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_at"),
+					resource.TestCheckResourceAttr(resourceName, "description", description),
+				),
+			},
+			{
+				Config: testAccAwsMacieClassificationJobconfigComplete(bucketName, macie2.JobStatusRunning, descriptionUpdated),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2ClassificationJobExists(resourceName, &macie2Output),
+					resource.TestCheckResourceAttr(resourceName, "job_type", macie2.JobTypeScheduled),
+					resource.TestCheckResourceAttr(resourceName, "job_status", macie2.JobStatusRunning),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_at"),
+					resource.TestCheckResourceAttr(resourceName, "description", descriptionUpdated),
+				),
+			},
```

Comment:
For these tests, since the structure is more complex, it would be worthwhile checking the nested values, such as `scoping` and its contents